llvm/test/CodeGen/RISCV/rvv/vsetvli-insert-coalesce.mir

```diff
@@ -0,0 +1,266 @@
+    $x13 = COPY undef %39
+    $x14 = COPY undef %55
+    $x15 = COPY undef %55
+    PseudoCALLIndirect undef %114, csr_ilp32d_lp64d, implicit-def dead $x1, implicit $x10, implicit $x11, implicit $x13, implicit $x14, implicit $x15, implicit-def $x2, implicit-def $x10
+    ADJCALLSTACKUP 0, 0, implicit-def dead $x2, implicit $x2
+    %57:gpr = COPY $x10
+    SW undef %114, undef %48, 0 :: (store (s32))
+    %14:gpr = PseudoCCMOVGPR undef %54, undef %57, 1, undef %114, undef %39
+    %15:gpr = PseudoCCMOVGPR undef %54, %57, 1, undef %114, undef %5
+    %16:gpr = ADD_UW undef %40, $x0
+    %17:gprnox0 = ADD_UW undef %30, $x0
+    %46:gpr = COPY undef %114
+
+  bb.3:
+    successors: %bb.5(0x04000000), %bb.4(0x7c000000)
+
+    %18:gpr = PHI undef %46, %bb.2, undef %93, %bb.10
+    %60:gpr = SLLI undef %18, 2
+    %19:gpr = OR undef %60, undef %14
+    %20:gpr = OR undef %60, undef %15
+    BEQ killed undef %18, $x0, %bb.5
+    PseudoBR %bb.4
+
+  bb.4:
+    successors: %bb.7(0x80000000)
+
+    %64:gpr = COPY $x0
+    %63:gpr = COPY undef %64
+    %62:gpr = COPY undef %64
+    PseudoBR %bb.7
+
+  bb.5:
+    successors: %bb.6(0x80000000)
+
+  bb.6:
+    successors: %bb.6(0x80000000)
+
+    PseudoBR %bb.6
+
+  bb.7:
+    successors: %bb.8(0x80000000)
+
+    %21:gpr = PHI undef %62, %bb.4, undef %111, %bb.9
+    %22:gpr = PHI undef %63, %bb.4, undef %28, %bb.9
+    ADJCALLSTACKDOWN 0, 0, implicit-def dead $x2, implicit $x2
+    $x10 = COPY undef %4
+    $x11 = COPY undef %1
+    $x12 = COPY undef %20
+    $x13 = COPY undef %19
+    PseudoCALLIndirect undef %114, csr_ilp32d_lp64d, implicit-def dead $x1, implicit $x10, implicit $x11, implicit $x12, implicit $x13, implicit-def $x2, implicit-def $x10
+    ADJCALLSTACKUP 0, 0, implicit-def dead $x2, implicit $x2
+    %68:gpr = PseudoMovAddr target-flags(riscv-hi) @fooo.g1, target-flags(riscv-lo) @fooo.g1
+    %23:gpr = SH3ADD %21, killed undef %68
+    %69:gpr = LUI target-flags(riscv-hi) @g0
+    %25:gprnox0 = PseudoVSETVLI %17, 199 /* e8, mf2, ta, ma */, implicit-def dead $vl, implicit-def dead $vtype
+    %65:vrm2 = PseudoVMV_V_I_M2 undef $noreg, 0, -1, 5 /* e32 */, 0 /* tu, mu */
+
+  bb.8:
+    successors: %bb.9(0x04000000), %bb.8(0x7c000000)
+
+    %26:vrm2nov0 = PHI %65, %bb.7, %70, %bb.8
+    %71:gpr = IMPLICIT_DEF
+    %72:gprnox0 = MINU undef %17, killed undef %71
+    %73:vrm4 = PseudoVLE16_V_M4 undef $noreg, undef %23, 1, 4 /* e16 */, 2 /* tu, ma */ :: (load unknown-size, align 32)
+    %74:gpr = IMPLICIT_DEF
+    early-clobber %84:vrm2nov0 = PseudoVLUXEI64_V_M4_M2_MASK undef $noreg, undef %69, killed undef %73, undef %115, killed undef %72, 5 /* e32 */, 1 /* ta, mu */ :: (load unknown-size, align 4)
+    %86:vrm2nov0 = PseudoVMERGE_VVM_M2 %26, undef %26, killed undef %84, killed undef %115, 0, 5 /* e32 */
+    %70:vrm2 = PseudoVMV_V_I_M2 undef $noreg, 1, -1, 5 /* e32 */, 0 /* tu, mu */
+    %27:vrm2 = COPY undef %86
+    BEQ killed undef %74, $x0, %bb.8
+    PseudoBR %bb.9
```

Comment:
Is it possible to trim this test case down to just block 7 and 8? Or is it relying on the other blocks to force a vsetvli toggle somewhere?